### DISCUSSION TITLE
How to show WPF controls top of WindowsFormsHost?

### DISCUSSION THREAD

**Discussion by VahidEra** [dotnet/wpf#7604] [Q&A] [answered]
Hi guys,
I have a WPF project that I have to use a WindowsFormsHost in, but unfortunately, WindowsFormsHost always rests on WPF controls.
But after hours of searching the internet and reading different articles and trying different ways even here on GitHub I still have the same problem and my problem is not solved. one of those articles is:[https://bartwullems.blogspot.com/2010/11/wpf-and-winforms-airspace-problem.html](url).
In my sample file, there is a host drawer on which WindowsFormsHost is placed. I will post the project file below, so you can take a look at it.
>***Is there a solution to this problem?***

My sample file is:
https://drive.google.com/file/d/1SCpkWRn8THhr1io-UW3h0i3kpI7nH67l/view?usp=sharing

**Comment by miloush** [comment, marked as answer]
No there isn't a solution and I wouldn't expect one any time soon, there already has been an attempt, I assume you have seen https://learn.microsoft.com/en-us/archive/blogs/dwayneneed/mitigating-airspace-issues-in-wpf-applications

If it is good enough for you, the article you refer to (as well as what Dwayne talks about) is taking a static bitmap of the winforms content and including it in WPF tree as a bitmap.

**Reply by VahidEra** [reply]
@miloush 
Dear John, first, I would like to thank you for helping me. Yes, you are absolutely right After checking the code, I realized that it takes a screenshot of the program and displays it, which in my opinion is a very useless and ridiculous task.
